src/fallback/dom_parser_implementation.cpp

```diff
@@ -243,6 +243,93 @@ WARN_UNUSED error_code implementation::minify(const uint8_t *buf, size_t len, ui
   // for fear of aliasing
   return SUCCESS;
 }
+WARN_UNUSED bool implementation::validate_utf8(const char *buf, size_t len) const noexcept { 
```

Comment:
I'm curious if we can use this for the stage 1 implementation as well :) My code is pretty naive ...

(Not something that should really be part of this PR.)